packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -72,14 +73,18 @@ class MojoAssetBundle extends AssetBundle {
     _imageCache = null;
   }
 
-  Future<ui.Image> _fetchImage(String key) async {
-    return await decodeImageFromDataPipe(await load(key));
+  Future<ui.Image> _fetchImage(String key, [ AssetResolver resolver ]) async {
+    if (resolver != null)
+      key = await resolver.resolve(key);
+    if (_imageCache.containsKey(key))
+      return _imageCache[key];
+    ui.Image value = await decodeImageFromDataPipe(await load(key));
+    _imageCache[key] = value;
+    return value;
   }
 
-  ImageResource loadImage(String key) {
-    return _imageCache.putIfAbsent(key, () {
-      return new ImageResource(_fetchImage(key));
-    });
+  ImageResource loadImage(String key, [ AssetResolver resolver ]) {
+    return new ImageResource(_fetchImage(key, resolver));
```

Comment:
why do you return a new one now? I thought the whole point of ImageResource was that if you fetched the same URL twice, you'd end up with the same resource wrapping the same future.